src/main/java/platform/codingnomads/co/springdata/lab/SpringDataLab.java

```diff
@@ -23,16 +29,70 @@ public static void main(String[] args) {
     @Override
     public void run(String... args) throws Exception {
 
-        final List<Area> areas = areaRepository.saveAll(
-                Arrays.asList(
-                        Area.builder().code("G").build(),
-                        Area.builder().code("H").build(),
-                        Area.builder().code("Y").build(),
-                        Area.builder().code("Z").build()
-                )
+        // Insert areas
+        final List<Area> areas = Arrays.asList(
+                Area.builder().code("G").build(),
+                Area.builder().code("H").build(),
+                Area.builder().code("Y").build(),
+                Area.builder().code("Z").build(),
+                Area.builder().code("A").build(),
+                Area.builder().code("B").build(),
+                Area.builder().code("C").build(),
+                Area.builder().code("D").build()
         );
+        for (Area area : areas) {  // For each area
+            if (areaRepository.findByCode(area.getCode()) == null) {  // Is it already in the DB?
+                areaRepository.save(area);  // If not, save it.
```

Comment:
It is not critical for this project but in general.
The data processing could be done either on DB side or in App: the tradeoff here is fast vs slow. 

DB engines are optimized to perform requests as efficiently and fast as possible given a correct request. Once the operation is finished, DB has to marshal all the data to a transport layer and send it to the app, where it would be unmarshaled, mapped to an object, and passed to logic for processing. 

Moving info through the transport layer is slow, as well as marshaling operations. The more info you need to move the slower it is. 

In this situation, all you need is to check if the entry exists because you ignore the operation result. So basically you need `boolean`, which is exactly one bit `0` or `1`. Also, if DB "knows" that the element itself is not required, but only if it exists, it will not load it from disk but just check, which is faster. 

So the possible more optimal solution could be: 
1. Use builtin JPA Repository `.exist(...)` operation like:
   ```java
    if (!areaRepository.existsById(1l); // if you have ID 
    if (!areaRepository.exists(Example.of(Area.builder().code("G").build()))); // if you don't have ID 
   ```
2. Add your method to the repository like:
   ```java
    boolean existsAreaByCode(String code);
    // and youse it 
    if(!areaRepository.existsAreaByCode("G"))
   ```

Reply:
Thank you!